asapdiscovery-simulation/asapdiscovery/simulation/utils.py

```diff
@@ -20,3 +29,148 @@ def create_protein_only_system(input_pdb_path: str, ff_params: ForceFieldParams)
+        alchem_result = AlchemiscaleResults(network_key=network_key)
+        network_data["results"] = alchem_result
+        # work around the frozen object
+        result_network = FreeEnergyCalculationNetwork(**network_data)
+        return result_network
+
+    def action_network(
+        self, planned_network: FreeEnergyCalculationNetwork
+    ) -> list[Optional[ScopedKey]]:
+        """
+        For the given network which is already stored on alchemiscale create and action tasks.
+
+        Args:
+            planned_network: The network which should action tasks for.
+
+        Returns:
+            A list of actioned tasks for this network.
+
+        """
+        network_key = planned_network.results.network_key
+
+        tasks = []
+        for tf_sk in self._client.get_network_transformations(network_key):
+            # for each task create x repeats which are configured by the factory
+            tasks.extend(
+                self._client.create_tasks(tf_sk, count=planned_network.n_repeats + 1)
+            )
+
+        # now action the tasks to ensure they are picked up by compute.
+        actioned_tasks = self._client.action_tasks(tasks, network_key)
+        return actioned_tasks
+
+    def network_status(
+        self, planned_network: FreeEnergyCalculationNetwork
+    ) -> dict[str, int]:
+        """
+        Get the status of the network from alchemiscale.
+
+        Args:
+            planned_network: The network which we should look up in alchemiscale.
+
+        Returns:
+            A dict of the status type and the number of instances.
+        """
+        network_key = planned_network.results.network_key
+        return self._client.get_network_status(network_key)
+
+    def collect_results(
+        self, planned_network: FreeEnergyCalculationNetwork
+    ) -> FreeEnergyCalculationNetwork:
+        """
+        Collect the results for the given network.
+
+        Args:
+            planned_network: The network who's results we should collect.
+
+        Returns:
+            A FreeEnergyCalculationNetwork with all current results. If any are missing and allow missing is false an error is raised.
+        """
+        # collect results following the notebook from openFE
+        results = []
+        for tf_sk in self._client.get_network_transformations(
+            planned_network.results.network_key
+        ):
+            raw_result = self._client.get_transformation_results(tf_sk)
+            if raw_result is None:
+                continue
+            # format into our custom result schema and save
+            estimate = raw_result.get_estimate()
+            uncertainty = raw_result.get_uncertainty()
+            # work out the name of the molecules and the phase of the calculation
+            individual_runs = list(raw_result.data.values())
+
+            if "protein" in individual_runs[0][0].inputs["stateA"].components:
```

Comment:
Could you add a note as to why this is required? I couldn't figure it out just by looking.